src/ray/protobuf/experimental/autoscaler.proto

```diff
@@ -271,6 +275,33 @@ message GetClusterStatusReply {
   ClusterResourceState cluster_resource_state = 2;
 }
 
+enum DrainNodeReason {
+  // The node has been idle for long enough time.
+  // Drain request with this reason is rejectable.
+  IDLE_TERMINATION = 0;
+  // The node will be preempted.
+  // Drain request with this reason is not rejectable.
+  PREEMPTION = 1;
+}
+
+message DrainNodeRequest {
+  // The id of the node to be drained.
+  bytes node_id = 1;
```

Comment:
are you referring to ray's internal node_id or instance_id (e.g. i-xxx)? Can we use instance_id for consistency?

Reply:
autoscaler should be also aware of ray's node id though

Reply:
Discussed offline: we will use the Ray node id.